### PR TITLE
Add compset support for northamericax4v1 atmos RRM grid

### PR DESCRIPTION
**Pull request by eroesler**
Adds compset support for northamericax4v1 atmos RRM grid

Adds configuration details to enable a component set for the atmosphere's northamericax4v1 RRM grid.  

[BFB]
[NML]

### CODE REVIEW

**Comment by rljacob** [comment]
@eroesler is this PR ready?  Please change title to describe what is being added/changed.

**Comment by tangq** [comment]
@eroesler , during a recent water cycle telecon, @golaz decided that the final grids for the v2 North America (NA) RRM are this atmosphere NA RRM grid, 1/8 degree land tri-grid (r0125), and the new ocean NA RRM grid.
As an intermedia step (before the ocean RRM grids are ready) for atmosphere-only test, we will use the oRRS18to6v3 ocean grid.
For the r0125 land grid, I got the mesh file from @jonbob, who is working on the tri-grid branch (https://github.com/E3SM-Project/E3SM/compare/jonbob/scripts/tri-grid). I will update the ocean and land files on this PR for the intermedia atmosphere-only test.

**Comment by tangq** [comment]
After a closer look at the atmosphere-only (F compset) settings, it appears to me that the ocean grid settings will be overwritten by the compset (e.g., F2010C5-CMIP6-HR) via the docn.streams.txt.prescribed file because data ocean is used for F cases. Therefore, we don't need to update the ocean-related settings on this PR.

**Comment by wlin7** [comment]
Great work, @tangq . You have also addressed all the comments by @brhillman on PR #3071 regarding using highorder map files for SMAPs and VMAPs. After this PR is tested and merged, we can choose to rebase #3071 or simply create a new branch for the Antarctica RRM.

**Comment by tangq** [comment]
@rljacob , "n3" has been removed from the file names.

**Comment by rljacob** [comment]
Ok.  Yes if the netcdf4 ones never appeared in a case on master you can just overwrite them with the netcdf3 ones and use the same name.

**Comment by tangq** [comment]
All the updated files are now on the input data repository at blues. When the tri-grid changes for r0125 #3249 are added, we can start to test this PR with F-cases.

**Comment by wlin7** [comment]
@tangq, a few notes on the file.

atm init file, atm/cam/inic/homme/cami_0001-01-01_northamericax4v1_c190709.nc, is not on the inputdata server yet. It is available on inputdata repo on NERSC. The data is generated cam.i after 1 year of simulation initialized from an IC derived from IFS analysis of 2016-08-01.

atm init and topo files are in 64bit_offset, OCN2ATM_FMAP and SMAP are in 64bit_data format.

**Comment by tangq** [comment]
@wlin7 , thanks for the suggestions.
The atm IC file will need to be recreated with the new land tri-grid. I will update it and upload the file to the data server later.
The OCN2ATM mapping files contain large variables, which requires 64-bit support. I used the "-5" option of ncks as suggested by @rljacob to convert these files. Accordingly to @rljacob , this format will work with Pnetcdf reader.

**Comment by tangq** [comment]
Thanks, @czender , for the prompt approval. It is the first time I used the ncremap TempestRemap options. The mapping files are exactly what I wanted your check to ensure they were done correctly.

**Comment by tangq** [comment]
To test this North America RRM on the tri-grid, it requires the tri-grid PR (#3249, which is ready to be merged) and the F2010C5-CMIP6-HR compset (currently only available on maint-1.0) on master.

@rljacob , I believe you added the F2010C5-CMIP6-LR compset, which is the low-res version of F2010C5-CMIP6-HR, to master from maint-1.0. Do you mind do the same for the F2010C5-CMIP6-HR compset? Or can you provide me some instructions on how to do it and I can give it a try. Thanks.

**Comment by rljacob** [comment]
I think I just did a copy and paste.   You can add what you need for testing to this branch.

You can rebase this branch to the trigrid branch if you want to test it with trigrid now.

**Comment by tangq** [comment]
@rljacob , I was wondering if I could use `git cherry-pick <commit-hash>` to add the changes on the F2010C5-CMIP6-HR branch to a new branch created from the current master. Did you used cherry-pick before?

**Comment by tangq** [comment]
After rebasing to today's master with tri-grid, this North America RRM built successfully and downloaded missing input files from the data server.

It failed at "Creating new decomp:" with errors "cam_filemap_get_filemap: internal error, pos"

My test was created with `--compset F2010SC5-CMIP6 --res northamericax4v1_r0125_oRRS15to5` on cori. Note that the dy-core settings from this compset are NOT correct, but it shouldn't fail the model at this stage.

The test is located at:/global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM/20191022_na.tst_NA_RRM.northamericax4v1_r0125_oRRS15to5.cori-knl

**Comment by tangq** [comment]
Some updates:
It seems that to the compset F2010SC5-CMIP6 work on tri-grid with MOSART river module, it needs to change to use `2010S_CAM5%CMIP6_CLM45%SPBC_CICE%PRES_DOCN%DOM_MOSART_SGLC_SWAV`.

After implementing this change, my test complains about missing input files for land. This makes sense as the F2010SC5-CMIP6 compset was created before tri-grid and there's no tri-grid input files for it.

**Comment by tangq** [comment]
It still complains about fatmlndfrc. Somehow, in Buildconf/clm.input_data_list fatmlndfrc is set to '/project/projectdirs/acme/inputdata/share/domains/UNSET', even when I changed it in components/cam/bld/namelist_files/namelist_defaults_cam.xml

Perhaps, @rljacob or @bishtgautam knows how fatmlndfrc is specified?

**Comment by wlin7** [comment]
I wonder with tri-grid, whether fatmlndfrc has to be set in ELM's namelist_defaults_clm4_5.xml instead of EAM's default file. It has been fine because atm and lnd on the same grid?

While waiting for expert to chime in, you may manually do the following in casedir to get the test going

./xmlchange LND_DOMAIN_FILE="domain.lnd.northamericax4v1_oRRS15to5.191022.nc"

**Comment by bishtgautam** [comment]
I believe the information about the new domain should be set in [cime/config/e3sm/config_grids.xml#L1645](https://github.com/E3SM-Project/E3SM/blob/master/cime/config/e3sm/config_grids.xml#L1645)

**Comment by tangq** [comment]
> I believe the information about the new domain should be set in [cime/config/e3sm/config_grids.xml#L1645](https://github.com/E3SM-Project/E3SM/blob/master/cime/config/e3sm/config_grids.xml#L1645)

@bishtgautam , this is what puzzles me. In the config_grids.xml, I already defined the domain files with the following lines. But it seems not working. Do you have any suggestions?

<domain name="ne0np4_northamericax4v1">
      <nx>130088</nx>
      <ny>1</ny>
      <file grid="atm|lnd" mask="oRRS15to5">$DIN_LOC_ROOT/share/domains/domain.lnd.northamericax4v1_oRRS15to5.191022.nc</file>
      <file grid="ice|ocn" mask="oRRS15to5">$DIN_LOC_ROOT/share/domains/domain.ocn.northamericax4v1_oRRS15to5.191022.nc</file>
      <desc>1-deg with 1/4-deg over North America (version 1):</desc>
    </domain>

**Comment by tangq** [comment]
> I wonder with tri-grid, whether fatmlndfrc has to be set in ELM's namelist_defaults_clm4_5.xml instead of EAM's default file. It has been fine because atm and lnd on the same grid?
> 
> While waiting for expert to chime in, you may manually do the following in casedir to get the test going
> 
> ./xmlchange LND_DOMAIN_FILE="domain.lnd.northamericax4v1_oRRS15to5.191022.nc"

This won't work as the model couldn't pass

**Comment by tangq** [comment]
Oh...hit the wrong button.

**Comment by tangq** [comment]
> I wonder with tri-grid, whether fatmlndfrc has to be set in ELM's namelist_defaults_clm4_5.xml instead of EAM's default file. It has been fine because atm and lnd on the same grid?
> 
> While waiting for expert to chime in, you may manually do the following in casedir to get the test going
> 
> ./xmlchange LND_DOMAIN_FILE="domain.lnd.northamericax4v1_oRRS15to5.191022.nc"

I added this xmlchange command in the run script as a temporary fix to get my tests going.

**Comment by tangq** [comment]
The test was submitted, but it only uses 2 nodes on cori-knl. I use the "L" option for processor_config in the run e3sm script. Is this a default layout? Feels too small for a RRM run.

**Comment by tangq** [comment]
The test failed at the same point at "Creating new decomp:" with errors "cam_filemap_get_filemap: internal error, pos" when initializing the atmosphere.

Can this be a PE-layout issue?

**Comment by tangq** [comment]
I think I figured why the LND_DOMAIN_FILE is not set. Testing now.

**Comment by tangq** [comment]
When ROF is turned off, the LND_DOMAIN_FILE not set issue disappears. I am not sure why this is happening though.

An issue was reported (https://github.com/E3SM-Project/E3SM/issues/3276) when trying to run the F20TRC5-CMIP6 compset on the ne30_r05_oECv3 grid. I noticed that ROF is ON for ne30_r05_oECv3. When ne30_r05_oECv3 works, I will try to mimic what it's done there for  northamericax4v1_r0125_oRRS15to5.

**Comment by tangq** [comment]
The LND_DOMAIN_FILE is set correctly after adding the land domain path under the r0125 domain. Turn on ROF works and finds the domain files.

**Comment by tangq** [comment]
Since the compset F20TRC5-CMIP6 is supported by the tri-grid and the test with F20TRC5-CMIP6 on ne30_r05_oECv3 (https://github.com/E3SM-Project/E3SM/issues/3276) ran further than my test, I decide to switch to test F20TRC5-CMIP6 on the northamericax4v1_r0125_oRRS15to5 grid and test F2010SC5-CMIP6 next.

The F20TRC5-CMIP6 test did pass the previous failing point of atm initialization and failed with out-of-memory errors during land initialization. @jonbob and @bishtgautam , did you need to make any changes for the ne30_r0125_oECv3? Of course, northamericax4v1_r0125_oRRS15to5 requires more memory, but I am curious if you had similar issues.

My test is at cori: /global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM/20191022_na.tst_NA_RRM.F20TRC5-CMIP6.northamericax4v1_r0125_oRRS15to5.cori-knl

**Comment by bishtgautam** [comment]
>The F20TRC5-CMIP6 test did pass the previous failing point of atm initialization and failed with out-of-memory errors during land initialization. 

Can you re-run the simulation with increased `NTASKS_LND`?

**Comment by tangq** [comment]
I am not sure what NTASKS numbers are appropriate, but I found the "customknl" layout settings in the run_e3sm script. Compared to the default 128 NTASKS for each component, the customknl asks 5400 NTASKS_ATM and 320 NTASKS_LND. It also lowers the MAX_TASKS_PER_NODE to 64 from 128. I am using it for the test (/global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM/20191022_na.tst_NA_RRM_PE.F20TRC5-CMIP6.northamericax4v1_r0125_oRRS15to5.cori-knl).

It now asks for 145 nodes instead of the default 2 nodes. I will post updates when it runs.

**Comment by tangq** [comment]
The OOM issue is solved by NTASKS_LND = 5400 (probably too many, but this is what NTASKS_ATM uses, so I use it for now).

The land module now failed to initialize with "ERROR in surfrdUtilsMod.F90 at line 61" because "read_variable_2d ERROR: sum of PCT_NAT_PFT not 1.0 at nl=           1"

The test is at: /global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM/20191022_na.tst_NA_RRM_PE.F20TRC5-CMIP6.northamericax4v1_r0125_oRRS15to5.cori-knl

@bishtgautam, since the tri-grid PR (#3249) supports ne30_r0125_ne30.F20TRC5-CMIP6, all the land files for r0125.F20TRC5-CMIP6 should be good. Do you have any ideas about why the model complains about it? Thanks!

**Comment by wlin7** [comment]
The error message during reading the landuse file contains

 Get data for variable PCT_NAT_PFT for year            0
and "sum is:  -3.784654473473011E+300".

The year is apparently not expected in this landuse time series from 1980 to 2015, and the sum must be due to many (if not all) missing data values.

I just took a look at the file 
/project/projectdirs/acme/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_0.125x0.125_hist_simyr1850-2015_c191004.nc

The time and YEAR values are all 0s after 1980. It seems data after year 1980 are missing. The landuse file needs to be examined.

@tangq, assuming the problem is due to missing data after 1980, you may try resetting RUN_STARTDATE="1979-01-01" to see if the model go further. Good luck!

**Comment by tangq** [comment]
Thanks, @wlin7 . I checked that in my test RUN_STARTDATE="2010-01-01". I changed it to 1979-01-01 as you suggested. Let's see if it makes a difference.

I am creating a confluence page summarizing the tests. I will post the link here later. It will be easier to know what's been tested on a confluence page than the messages here.

**Comment by tangq** [comment]
I created a confluence page (https://acme-climate.atlassian.net/wiki/spaces/ED/pages/1045202354/Tri-grid+tests+for+F2010SC5-CMIP6+compset) to keep track of the tests.

In short, I tried to reproduce the supported tri-grid case ne30_r0125_ne30.F20TRC5-CMIP6, but it seems failing because of the land use file issue (see @wlin7 's comments above).

**Comment by tangq** [comment]
@wlin7 , do you know how the PE-layout is picked up by different resolutions? I tried `--pecount L` with two resolutions and the layouts are very different on cori-knl.

`--res ne30_r0125_ne30 --pecount L`: 85 nodes
`--res northamericax4v1_r0125_oRRS15to5 --pecount L`: 2 nodes

Perhaps, this is set by CIME and @rljacob knows more.

**Comment by wlin7** [comment]
For pelayout, it could be. Often during testing, I would just manually set the MPI ntasks in the run script using the following csh script

  ```
set npes=800   
  foreach comp ( CPL ATM LND OCN ICE ROF WAV GLC ) 
     ./xmlchange NTASKS_$comp="$npes"
 end
  ./case.setup --reset
```

**Comment by bishtgautam** [comment]
@tangq I believe the `landuse.timeseries_0.125x0.125_hist_simyr1850-2015_c191004.nc` wasn't download correctly on Cori as its size is 23G, while the file on e3sm inputdata server is 119G

On Cori
```
ls -lh /project/projectdirs/acme/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_0.125x0.125_hist_simyr1850-2015_c191004.nc
-rw-rw-r--+ 1 tang30 tang30 23G Nov  1 12:35 /project/projectdirs/acme/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_0.125x0.125_hist_simyr1850-2015_c191004.nc
```
On Blues
```
ls -lh /lcrc/group/acme/public_html/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_0.125x0.125_hist_simyr1850-20
-rw-r--r--+ 1 gbisht collab 119G Oct 10 11:28 /lcrc/group/acme/public_html/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_0.125x0.125_hist_simyr1850-2015_c191004.nc
```
I didn't want to overwrite the file on NERSC, so I copied the new file in `/project/projectdirs/acme/inputdata/lnd/clm2/surfdata_map/0.125x0.125/`. The F20TRC5-CMIP6.ne30_r0125_ne30 with `RUN_STARTDATE=2010-01-01` ran successfully with the new file. If you can successfully run a simulation with the new file, we can overwrite the old file.

**Comment by wlin7** [comment]
Thanks for checking, @bishtgautam . The file you copied over does have the complete time series. 
Depending on how/where the file was truncated, the test run with RUN_STARTDATE="1979-01-01" may also fail. The new file should fix this problem.

**Comment by tangq** [comment]
@wlin7 , @bishtgautam , thanks for the help.

With the complete land use file, the F20TRC5-CMIP6.ne30_r0125_ne30 tests successfully ran 1 day with RUN_STARTDATE="1979-01-01" and RUN_STARTDATE="2010-01-01".

Now, I am moving to test F2010SC5-CMIP6.ne30_r0125_ne30.

**Comment by tangq** [comment]
F2010SC5-CMIP6.ne30_r0125_ne30 is successful too. I will compare what are different between ne30_r0125_ne30 and northamericax4v1_r0125_oRRS15to5.

**Comment by tangq** [comment]
I am trying to figure out why F2010SC5-CMIP6.northamericax4v1_r0125_oRRS15to5 failed at atm initialization by comparing to the 1-day successful case of F2010SC5-CMIP6.ne30_r0125_ne30.

My tests on cori are under /global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM:
F2010SC5-CMIP6.northamericax4v1_r0125_oRRS15to5:
20191022_na.tst_NA_RRM_PE_compset.northamericax4v1_r0125_oRRS15to5.cori-knl/
F2010SC5-CMIP6.ne30_r0125_ne30:
20191022_na.tst_NA_RRM.F2010SC5-CMIP6.ne30_r0125_ne30.cori-knl/

But the *_in files in the directories above seem reasonable to me.

I also checked the grid settings in cime/config/e3sm/config_grids.xml and found no obvious differences in the way northamericax4v1_r0125_oRRS15to5 and ne30_r0125_ne30 are set up. @wlin7 , @bishtgautam , @jonbob , do you have any suggestions?

**Comment by wlin7** [comment]
Hi @tangq , by back tracing the codes, it turns out the error occur during reading initial data for U, which is the first state variable to read. Growing suspicious of it, I checked and found that ne30 IC file was used in effect. This is unintended. The attribute setting for ncdata likely not properly resolved.

**Comment by wlin7** [comment]
The problem is that the ncdata (IC), which is for ne30,  is hardwired for compset F2010C5-CMIP6 in the use_case file. Settings in use_case file overrides others. The compset can't be directly used for this RRM without change. I am doing a quick test by specifying the intended ncdata on user_n_cam.

**Comment by wlin7** [comment]
A further note on the use_case file. F2010SC5-CMIP6 was created specifically for ne30. If to make it more generic that is usable for other grids, can expand the ncdata section in the similar format as in namelist_defaults_cam.xml. Adding the following to 2010S_cam5_CMIP6.xml does allow the RRM IC file to be picked up correctly.

<ncdata dyn="se" hgrid="ne0np4_northamericax4v1" nlev="72"    >atm/cam/inic/homme/cami_0001-01-01_northamericax4v1_c190709.nc</ncdata>

**Comment by tangq** [comment]
@wlin7 , good catch. I am always a bit confused about how those xml settings are being implemented. It looks like the compset.xml will overwrite the namelist_defaults_cam.xml settings. This makes sense.

To avoid this issue in the future, it should be noted somewhere that the resolution-specific settings should be set in namelist_defaults_cam.xml. I will modify the F2010SC5-CMIP6 compset.

**Comment by tangq** [comment]
A second thought...It seems better to keep the ncdata settings in the compset xml file, because multiple compsets (e.g., F20TR and F2010) will need to run on the same resolution but with different ncdata files. This is different than for example topo data files which are shared by different compsets on the same resolution.

I feel it is better to keep the compset-dependent settings in the compset xml file, while keep the resolution-dependent ones in the namelist_defaults_cam.xml file. What do you think, @wlin7 ?

**Comment by tangq** [comment]
After fixing the ncdata issue with the F2010SC5-CMIP6 compset, the F2010SC5-CMIP6.northamericax4v1_r0125_oRRS15to5 test successfully ran 77 time steps and died due to time out on debug queue.

The test is at cori: 
/global/cscratch1/sd/tang30/E3SM_simulations/tst_NA_RRM/20191022_na.F2010SC5-CMIP6_PE.northamericax4v1_r0125_oRRS15to5.cori-knl

This PR looks good to go now, except it picks up a 2-node layout with `--pecount L`. Is there a way to make it default to a larger layout?

The F2010SC5-CMIP6 compset changes will be done in a separate PR.

**Comment by rljacob** [comment]
@ndkeen can you see why the test above only gets 2 nodes?

**Comment by ndkeen** [comment]
I don't think we have any compsets catching ` ATM_GRID is ne0np4_northamericax4v1` for resolution and the compset: `2010S_CAM5%CMIP6_CLM45%SPBC_CICE%PRES_DOCN%DOM_MOSART_SGLC_SWAV_SIAC_SESP` is missing `.+MPASSI.+MPASO` to be caught by some of the coupled stanzas.  I'm thinking we would need to look at performance for these as a whole? Or can we assume they will perform similarly to our "coupled lowres"?  Would we want to update this PR to adjust config_pesall.xml or make another one?

**Comment by rljacob** [comment]
Update this PR.   A later PR can find optimized layouts but this one should at least do more the 2 as an out-of-the-box config.   You can start having it use as many as the ne30 layouts.

**Comment by tangq** [comment]
The ne0np4_northamericax4v1 ATM grid will be added as a part of this PR. It is the official v2 RRM grid. For now, we just need some reasonable layout to perform tests for this PR.

Later, as @rljacob suggested we will optimize the layout for routine use of this resolution in a separate PR.

**Comment by rljacob** [comment]
@tangq what is the --res and --compset you want to test and get useful layouts for?

**Comment by tangq** [comment]
I am testing --res northamericax4v1_r0125_oRRS15to5 --compset F2010SC5-CMIP6. This setting will be a major one for v2 high-res atmosphere development once my tests are successful.

**Comment by rljacob** [comment]
@tangq is this PR ready?

**Comment by tangq** [comment]
@rljacob , we are not lucky yet...My tests fail in 1 day with either the domain check or the aerosol optical depth too large issue. I will post updates when we make progresses.

**Comment by rljacob** [comment]
The purpose of this PR is to make it possible to configure a northamerican RRM case.  If it does that, it should be merged.   Finding the right settings to make it run can be a later PR.

**Comment by tangq** [comment]
@rljacob , I made some clean up to this PR. It can configure a North American RRM case. It is ready to be merged.

**Comment by rljacob** [comment]
@wlin7 please start merging this.

**Comment by wlin7** [comment]
Yes, will do this afternoon. Thanks @tangq for working this out.

**Comment by wlin7** [comment]
Reviewed one more time. All look good. Merged to next. Don't see any existing nightly test could have an NML diff.

**Comment by tangq** [comment]
@wlin7 , thanks for double checking.

**Comment by rljacob** [comment]
@wlin7 finish this PR.

**Comment by wlin7** [comment]
Thanks for the reminder @rljacob . Just merged to master. I have been watching. Not receiving the regression status report for a few days. This PR should be fine as it has been confirmed not breaking any next tests.

**Comment by ndkeen** [comment]
I'm looking at improving the PE layouts, but will make a new PR.
What is the test to use?

`create_test SMS.F2010SC5-CMIP6.northamericax4v1_r0125_oRRS15to5` ?

**Comment by tangq** [comment]
We need the layout for F-cases, so `F2010SC5-CMIP6.northamericax4v1_r0125_northamericax4v1`. Thanks for working on this, @ndkeen .

**Comment by ndkeen** [comment]
I tried this:

```
cori04% create_test F2010SC5-CMIP6.northamericax4v1_r0125_northamericax4v1
ERROR: Could not fill-out test name, partial string 'F2010SC5-CMIP6.northamericax4v1_r0125_northamericax4v1' had no compset information and you did not provide any
```

I also tried using create_newcase with `--res northamericax4v1_r0125_oRRS15to5 --compset F2010SC5-CMIP6` (as above), but then it failed at case.setup:

`ERROR: Need to provide valid mapping file between atm and rof in xml variable atm2rof_fmapname`

**Comment by tangq** [comment]
We don't have a test for this resolution yet.

In the create_newcase command, use `northamericax4v1_r0125_northamericax4v1` instead of `northamericax4v1_r0125_oRRS15to5`. Let me know if it still doesn't work.

**Comment by ndkeen** [comment]
OK I tried:
```
create_newcase --case /global/cscratch1/sd/ndk/acme_scratch/cori-knl/mfrrm/f.na.mfrrm.n020a64x1 --res northamericax4v1_r0125_northamericax4v1 --mach cori-knl --compiler intel --mpilib=mpt --compset F2010SC5-CMIP6 --project acme --walltime=0:26:00 --queue=debug
```

But in case.setup I get: `ERROR: Need to provide valid mapping file between atm and rof in xml variable atm2rof_fmapname`

File: cime/config/e3sm/config_grids.xml [hunk 1]
```diff
@@ -834,6 +834,16 @@
...
     </model_grid>
 
+    <model_grid alias="northamericax4v1_r0125_oRRS15to5" compset="(DOCN|XOCN|SOCN|AQP1)">
+      <grid name="atm">ne0np4_northamericax4v1</grid>
+      <grid name="lnd">r0125</grid>
+      <grid name="ocnice">oRRS15to5</grid>
+      <grid name="rof">r0125</grid>
+      <grid name="glc">null</grid>
+      <grid name="wav">null</grid>
+      <mask>oRRS15to5</mask>
```
[the remark this answers is not in the document]
Comment (tangq) [reply cime/config/e3sm/config_grids.xml]: Hi @jonbob , when @golaz assigned me to work on this PR, we planned to do it as two steps. 1 - NA RRM F-case grid (northamericax4v1_r0125_northamericax4v1) 2 - NA RRM coupled grid.

The F-case grid works now. The coupled grid is still waiting for the ocean RRM grid. While waiting for the ocean grid, we use the coupled northamericax4v1_r0125_oRRS15to5 grid as a test to apply the NA atmosphere RRM grid for coupled simulations. The northamericax4v1_r0125_oRRS15to5 grid won't be used in real simulations. The final goal is to replace oRRS15to5 with the ocean RRM grid. We will update the settings when the ocean RRM grid is available in a future PR.

**Comment by tangq** [comment]
@ndkeen , I am testing now...

**Comment by tangq** [comment]
It seems that the `atm2rof_fmapname` is recently added.

Screen Shot 2019-11-27 at 3 05 12 PM

It should only be required for "_MOSART" compset. I am not sure why `F2010SC5-CMIP6` has this requirement.

**Comment by tangq** [comment]
@ndkeen , I created 2 PRs (#3287 and #3331 ) for F2010SC5-CMIP6.northamericax4v1_r0125_northamericax4v1 to use the latest settings.

I tested that after deleting the two ATM2ROF lines above, my case setup and built without problems. This confirms that the issue is caused by these two lines. But, I don't know enough about xml parsing to fix it. Any suggestions? I can add the fix to #3331 .

**Comment by rljacob** [comment]
Those atm2rof lines were added in #3133.  @hydrotian @liho745 were you supposed to add those to F2010SC5-CMIP6?

**Comment by tangq** [comment]
These atm2rof maps are not required for F-cases before. If they are supposed to be, then all the F compsets need to be updated.

**Comment by tangq** [comment]
My test using the code without the two atm2rof lines eventually failed at (prep_rof_init) : Initializing mapper_Fa2r. (Fa2r stands for fluxes between atmosphere and river?)

I checked my successful test done a couple of days ago (11/23). Such initialization of mapper_Fa2r doesn't exist in that test.

We need a way to turn off coupling between MOSART-heat and EAM for F compsets. But I am not sure how to do it.

**Comment by liho745** [comment]
@jonbob, I feel you might be the best person to help with @tangq's request. Would you have some time for it?

**Comment by bishtgautam** [comment]
Try changing [cime_comp_mod.F90#L1586](https://github.com/E3SM-Project/E3SM/blob/51c291b12d76fe32376e9559cc309c1015a1cfa4/cime/src/drivers/mct/main/cime_comp_mod.F90#L1586) to
`if (rof_prognostic .and. (.not. atm_prognostic)) atm_c2_rof = .true.`

**Comment by tangq** [comment]
@liho745 , can you try @bishtgautam 's suggestion above? I don't know enough about your changes in #3133 to test.

**Comment by rljacob** [comment]
Why does F2010SC5-CMIP6 have MOSART active?  All the other F-cases are using the stub runoff model.

**Comment by rljacob** [comment]
@liho745 is there ever a reason to use mosart-heat without an active ocean model?

**Comment by bishtgautam** [comment]
As far as I understand, mosart-heat has been only tested for I-compsets

**Comment by hydrotian** [comment]
I don't think F-cases need to turn on MOSART-heat. @jonbob added mapper_Fa2r in #3133 to pass energy fluxes from ATM to ROF. I think he is the right person to help make it only available for I-cases?

**Comment by rljacob** [comment]
The problem is that F-cases shouldn't use MOSART at all.  @golaz added F20TRC5-CMIP6 with MOSART and @tangq added F2010SC5-CMIP6 with MOSART.  Was that intentional?

**Comment by golaz** [comment]
My understanding is that we had decided that moving forward, F cases would include MOSART to facilitate @hydrotian 's work.

As of now, #3133 is preventing progress on tri-grid. Since MOSART heat is not a v2 feature, I think it would be preferable to take it out for now.

**Comment by hydrotian** [comment]
@golaz @liho745 I thought MOSART-heat is one of the V2 BGC features. But since I didn't attend the all-hands meeting this time, I might be missing something...

**Comment by lrleung** [comment]
I’ve discussed the readiness of MOSART-heat for v2 with Hongyi. We don’t think it’s ready for two-way coupling within E3SM because more work is needed to ensure energy conservation. We need MOSART-heat for the v2 BGC experiments, but we can do offline modeling for the science question. So I think it’s reasonable to drop MOSART-heat in the tri-grid configuration. I’m copying Hongyi in case he has other suggestions.
-Ruby

**Comment by rljacob** [comment]
@hydrotian do you need cases that have a full atmosphere with mosart?

**Comment by hydrotian** [comment]
@rljacob without heat (#3133 ), MOSART doesn't need any input from ATM. With heat implemented, MOSART will need ATM fluxes either from full atmosphere or DATM. So if we've decided to leave heat offline, I don't think we need cases that run full atmosphere with MOSART.

**Comment by rljacob** [comment]
Ok lets make the fix that @bishtgautam recommended in https://github.com/E3SM-Project/E3SM/pull/3067#issuecomment-559803720 .  That means you can run an F-case with MOSART but without MOSART-heat and will get things working again.  The correct fix will be to tell the coupler the value of heatflag (through the infobuffer) and use that to set atm_c2_rof

**Comment by tangq** [comment]
Besides @bishtgautam 's suggestion in [#3067 (comment)](https://github.com/E3SM-Project/E3SM/pull/3067#issuecomment-559803720), this xml parsing issue [#3067 (xml parsing)](https://github.com/E3SM-Project/E3SM/pull/3067#issuecomment-559280634) has to be fixed to ONLY require ATM2ROF maps for the MOSART compsets.

**Comment by rljacob** [comment]
@tangq  won't Gautum's fix also remove the requirement in practice?  That map won't actually be checked for at runtime if atm_c2_rof is false.

**Comment by tangq** [comment]
@rljacob , Gautum's fix removes the requirement at runtime, but the xml parsing issue prevents F-cases being created (see @ndkeen 's [comment](https://github.com/E3SM-Project/E3SM/pull/3067#issuecomment-559275657)).

**Comment by jonbob** [comment]
@tangq - I think it will fix the problem @ndkeen saw. Otherwise no F-cases would be working with master. Part of case setup also checks for required files

**Comment by tangq** [comment]
Umm...I had to remove those two ATM2ROF map lines in my test, otherwise `create_newcase` would fail with the same errors @ndkeen reported.

**Comment by rljacob** [comment]
There are only 2 F cases with MOSART and neither of them are tested.

**Comment by rljacob** [comment]
Most F-cases use SROF.

**Comment by tangq** [comment]
The F2010SC5-CMIP6 compset, which will be heavily used for v2 atmosphere development, is one of the two with MOSART.

**Comment by bishtgautam** [comment]
Definition of all F cases that use MOSART should be changed to use SROF

**Comment by rljacob** [comment]
@bishtgautam see https://github.com/E3SM-Project/E3SM/pull/3067#issuecomment-560934331

**Pull request merged**